### ISSUE TITLE
Typescript 4.8.2 does not like the type definitions from sequelize 6.21.4

### ISSUE THREAD

**Issue by dilyanpalauzov** [sequelize/sequelize#14934]
Upgrading to Typescripte 4.8.2 and running tsc prints on my system:
```
node_modules/sequelize/types/hooks.d.ts:69:19 - error TS2344: Type 'TAttributes' does not satisfy the constraint '{}'.                  
                                                                                                                                        
69   M extends Model<TAttributes, TCreationAttributes> = Model,                                                                         
                     ~~~~~~~~~~~                                                                                                        
                                                                                                                                        
  node_modules/sequelize/types/hooks.d.ts:70:3                                                                                          
    70   TAttributes = any,                                                                                                             
         ~~~~~~~~~~~~~~~~~                                                                                                              
    This type parameter might need an `extends {}` constraint.                                                                          
                                                                                                                                        
node_modules/sequelize/types/model.d.ts:3252:105 - error TS2344: Type 'TModelAttributes' does not satisfy the constraint '{}'.          
                                                                                                                                        
3252 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCreation
Attributes>;
                                                                                                             ~~~~~~~~~~~~~~~~

  node_modules/sequelize/types/model.d.ts:3252:23
    3252 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCrea
tionAttributes>;
                               ~~~~~~~~~~~~~~~~~~~~~~
    This type parameter might need an `extends {}` constraint.

node_modules/sequelize/types/model.d.ts:3260:52 - error TS2344: Type 'S' does not satisfy the constraint '{}'.

3260 export type ModelDefined<S, T> = ModelStatic<Model<S, T>>;
                                                        ~

  node_modules/sequelize/types/model.d.ts:3260:26
    3260 export type ModelDefined<S, T> = ModelStatic<Model<S, T>>;
                                  ~
    This type parameter might need an `extends {}` constraint.
```
As these errors do not reference my files, I assume that the files with type definitions exported by sequelize 6.21.4 are not good.

**Comment by WikiRik** [comment]
Thanks for letting us know! We have not looked into TypeScript 4.8 support yet, but it will land in the next release.

**Comment by benasher44** [comment]
Yep getting same issue here. The fix should just be to switch the constraint to `extends {} = {}` anywhere where the generic contraint is ` = {}` (tells TypeScript you don't want `null` or `undefined`)

**Comment by WhiteMinds** [comment]
The conflict comes from https://github.com/microsoft/TypeScript/pull/49119

**Comment by donleistman** [comment]
Same issue here

**Comment by damlatato** [comment]
we are facing the same issue

**Comment by ephys** [comment]
You may be having a different issue because this was released in Sequelize 6.23 & 7.0.0-alpha.17
